File: packages/expo-manifests/ios/EXManifests/EXManifestsManifestFactory.m
//  Copyright © 2021 650 Industries. All rights reserved.

#import <EXManifests/EXManifestsManifestFactory.h>
#import <EXManifests/EXManifestsLegacyManifest.h>
#import <EXManifests/EXManifestsNewManifest.h>
#import <EXManifests/EXManifestsBareManifest.h>

@implementation EXManifestsManifestFactory

+ (nonnull EXManifestsManifest *)manifestForManifestJSON:(nonnull NSDictionary *)manifestJSON {
  EXManifestsManifest *manifest;
  if (manifestJSON[@"releaseId"]) {
    manifest = [[EXManifestsLegacyManifest alloc] initWithRawManifestJSON:manifestJSON];
  } else if (manifestJSON[@"metadata"]) {
    manifest = [[EXManifestsNewManifest alloc] initWithRawManifestJSON:manifestJSON];
  } else {
    manifest = [[EXManifestsBareManifest alloc] initWithRawManifestJSON:manifestJSON];
  }
  return manifest;
}

@end

File: packages/expo-manifests/ios/EXManifests/EXManifestsManifestFactory.swift
//  Copyright © 2021 650 Industries. All rights reserved.

import Foundation

@objcMembers
public class EXManifestsManifestFactory: NSObject {
  public static func manifest(forManifestJSON: [String: Any]) -> EXManifestsManifest {
    if forManifestJSON["releaseId"] != nil {
      return EXManifestsLegacyManifest(rawManifestJSON: forManifestJSON)
    } else if forManifestJSON["metadata"] != nil {
      return EXManifestsNewManifest(rawManifestJSON: forManifestJSON)
    } else {
      return EXManifestsBareManifest(rawManifestJSON: forManifestJSON)
    }
  }
}
